goldigen/generator_test.go

```diff
@@ -49,11 +50,33 @@ var _ = Describe("Generator", func() {
 	)
 
 	BeforeEach(func() {
-		config := main.NewConfig(outputPackageName, "RegisterTypes", inputPath, outputPath)
-		gen = main.NewGenerator(config)
+		config := NewConfig(outputPackageName, "RegisterTypes", inputPath, outputPath)
+		gen = NewGenerator(config)
 		output = &bytes.Buffer{}
 	})
 
+	It("should return error for invalid definition", func() {
+		yaml := `
+			types:
+				goldi.test.foo:
+					package: test
+					factory: @foo_provider::NewFoo
+					args:
+						invalid yml`
+
+		Expect(gen.Generate(strings.NewReader(yaml), output)).To(MatchError("could not parse type definition: yaml: unmarshal errors:\n  line 6: cannot unmarshal !!str `invalid...` into []interface {}"))
```

Comment:
The test is good but the matched error message is a bit too specific for my taste since it relies on the exact error string of the YAML library.

I would change the test slightly like this:
```suggestion
		Expect(err).To(HaveOccurred())
		Expect(err.Error()).To(HavePrefix("could not parse type definition: yaml: unmarshal errors:"))
```

Reply:
Agree.